src/qt/pivx/settings/settingsinformationwidget.cpp

```diff
@@ -143,8 +153,15 @@ void SettingsInformationWidget::setNumConnections(int count){
 
 void SettingsInformationWidget::setNumBlocks(int count){
     ui->labelInfoBlockNumber->setText(QString::number(count));
-    if (clientModel)
+    if (clientModel) {
         ui->labelInfoBlockTime->setText(clientModel->getLastBlockDate().toString());
+        ui->labelInfoBlockHash->setText(clientModel->getLastBlockHash());
```

Comment:
As said in the meeting, the concept is good. 

Just two issues with this two lines:
The `getLastBlockHash` and `getLastBlockDate` method lock `cs_main` which must never be locked or tried to lock directly from the main UI thread. Any waiting for lock period will freeze the whole wallet.